src/addresstype.h

```diff
@@ -90,6 +90,19 @@ struct WitnessV1Taproot : public XOnlyPubKey
     explicit WitnessV1Taproot(const XOnlyPubKey& xpk) : XOnlyPubKey(xpk) {}
 };
 
+struct PayToAnchor
+{
+    PayToAnchor() = default;
+
+    friend bool operator==(const PayToAnchor& a1, const PayToAnchor& a2) {
+        return true;
+    }
+
+    friend bool operator<(const PayToAnchor& a1, const PayToAnchor& a2) {
+        return false;
+    }
+};
```

Comment:
Is there something that will prevent us from deriving `PayToAnchor` from `WitnessUnknown`?

```c++
struct PayToAnchor: public WitnessUnknown
{
    PayToAnchor() : WitnessUnknown(1, {0x4e, 0x73}) {};
    friend bool operator==(const PayToAnchor& a1, const PayToAnchor& a2) {
        return true;
    }
    friend bool operator<(const PayToAnchor& a1, const PayToAnchor& a2) {
        return true;
    }
};
```

This way we can reuse the `WitnessUnknown` operator() that should be reused.

<details>
<summary>diff</summary>

```diff
diff --git a/src/addresstype.cpp b/src/addresstype.cpp
index bfcc6cc62a4..1b22484c85b 100644
--- a/src/addresstype.cpp
+++ b/src/addresstype.cpp
@@ -143,11 +143,6 @@ public:
         return CScript() << OP_1 << ToByteVector(tap);
     }
 
-    CScript operator()(const PayToAnchor& anchor) const
-    {
-        return CScript() << OP_1 << std::vector<unsigned char>{0x4e, 0x73};
-    }
-
     CScript operator()(const WitnessUnknown& id) const
     {
         return CScript() << CScript::EncodeOP_N(id.GetWitnessVersion()) << id.GetWitnessProgram();
diff --git a/src/addresstype.h b/src/addresstype.h
index ea59578f760..fdb2230d8ed 100644
--- a/src/addresstype.h
+++ b/src/addresstype.h
@@ -90,18 +90,6 @@ struct WitnessV1Taproot : public XOnlyPubKey
     explicit WitnessV1Taproot(const XOnlyPubKey& xpk) : XOnlyPubKey(xpk) {}
 };
 
-struct PayToAnchor
-{
-    PayToAnchor() = default;
-
-    friend bool operator==(const PayToAnchor& a1, const PayToAnchor& a2) {
-        return true;
-    }
-
-    friend bool operator<(const PayToAnchor& a1, const PayToAnchor& a2) {
-        return false;
-    }
-};
 
 //! CTxDestination subtype to encode any future Witness version
 struct WitnessUnknown
@@ -129,6 +117,18 @@ public:
     }
 };
 
+
+struct PayToAnchor: public WitnessUnknown
+{
+    PayToAnchor() : WitnessUnknown(1, {0x4e, 0x73}) {};
+    friend bool operator==(const PayToAnchor& a1, const PayToAnchor& a2) {
+        return true;
+    }
+    friend bool operator<(const PayToAnchor& a1, const PayToAnchor& a2) {
+        return true;
+    }
+};
+
 /**
  * A txout script categorized into standard templates.
  *  * CNoDestination: Optionally a script, no corresponding address.
diff --git a/src/key_io.cpp b/src/key_io.cpp
index 6e869de8a5b..bba10011c1f 100644
--- a/src/key_io.cpp
+++ b/src/key_io.cpp
@@ -64,16 +64,6 @@ public:
         ConvertBits<8, 5, true>([&](unsigned char c) { data.push_back(c); }, tap.begin(), tap.end());
         return bech32::Encode(bech32::Encoding::BECH32M, m_params.Bech32HRP(), data);
     }
-
-    std::string operator()(const PayToAnchor& anchor) const
-    {
-        std::vector<unsigned char> data = {1};
-        std::vector<unsigned char> program = {0x4e, 0x73};
-        data.reserve(1 + (2 * 8 + 4) / 5);
-        ConvertBits<8, 5, true>([&](unsigned char c) { data.push_back(c); }, program.begin(), program.end());
-        return bech32::Encode(bech32::Encoding::BECH32M, m_params.Bech32HRP(), data);
-    }
-
     std::string operator()(const WitnessUnknown& id) const
     {
         const std::vector<unsigned char>& program = id.GetWitnessProgram();
``` 
</details>

Reply:
done, with even fewer lines. Nice to reduce the boilerplate a bit.